src/kontrol/foundry.py

```diff
@@ -550,7 +553,91 @@ def foundry_show(
             res_lines += print_failure_info(proof, kcfg_explore, counterexample_info)
             res_lines += Foundry.help_info()
 
-    return '\n'.join(res_lines)
+    if to_kevm_claims:
+
+        def _remove_foundry_config(_cterm: CTerm) -> CTerm:
+            kevm_config_pattern = KApply(
+                '<generatedTop>',
+                [
+                    KApply('<foundry>', [KVariable('KEVM_CELL'), KVariable('CHEATCODES_CELL')]),
+                    KVariable('GENERATEDCOUNTER_CELL'),
+                ],
+            )
+            kevm_config_match = kevm_config_pattern.match(_cterm.config)
+            if kevm_config_match is None:
+                _LOGGER.warning('Unable to match on <kevm> cell.')
+                return _cterm
+            return CTerm(kevm_config_match['KEVM_CELL'], _cterm.constraints)
+
+        def _contains_foundry_klabel(_kast: KInner) -> bool:
+            _contains = False
+            _foundry_labels = [
+                '#call_foundry___FOUNDRY-CHEAT-CODES_KItem_Int_Bytes',
+                '#error_foundry___FOUNDRY-CHEAT-CODES_KItem_Int_Bytes',
+                '#return_foundry___FOUNDRY-CHEAT-CODES_KItem_Int_Int',
+                'FOUNDRY_WHITELISTCALL_FOUNDRY-CHEAT-CODES_ExceptionalStatusCode',
+                'FOUNDRY_WHITELISTSTORAGE_FOUNDRY-CHEAT-CODES_ExceptionalStatusCode',
+                '#updateRevertOutput___FOUNDRY-CHEAT-CODES_KItem_Int_Int',
+                '#checkRevert_FOUNDRY-CHEAT-CODES_KItem',
+                '#clearExpectRevert_FOUNDRY-CHEAT-CODES_KItem',
+            ]
+
+            def _collect_klabel(_k: KInner) -> None:
+                nonlocal _contains
+                if type(_k) is KApply and _k.label.name in _foundry_labels:
+                    _contains = True
+
+            collect(_collect_klabel, _kast)
+            return _contains
+
+        for node in proof.kcfg.nodes:
+            proof.kcfg.replace_node(node.id, _remove_foundry_config(node.cterm))
+
+        # Due to bug in KCFG.replace_node: https://github.com/runtimeverification/pyk/issues/686
+        proof.kcfg = KCFG.from_dict(proof.kcfg.to_dict())
+
+        _module_name = (
+            proof.id.upper()
+            .replace('%', '-')
+            .replace('.', '-')
+            .replace('(', '-')
+            .replace(')', '-')
+            .replace(':', '-')
+            .replace(',', '-')
+        )
+        _module_name += '-SPEC'
+        module_name = ''
+        is_hyphen = False
+        for char in _module_name:
+            if char == '-':
+                if not is_hyphen:
+                    module_name += char
+                is_hyphen = True
+            else:
+                is_hyphen = False
+                module_name += char
```

Comment:
Here you could use `re` instead.
```suggestion
        _module_name = re.sub(r'[%().:,]', '-', proof.id.upper()) + '-SPEC'
        module_name = re.sub(r'-+', '-', _module_name)
```

Reply:
Nice idea, done!